src/Realm.C

```diff
@@ -2085,10 +2097,15 @@ Realm::create_mesh()
   }
 
   // Initialize meta data (from exodus file); can possibly be a restart file..
-  inputMeshIdx_ = ioBroker_->add_mesh_database( 
-   inputDBName_, restarted_simulation() ? stk::io::READ_RESTART : stk::io::READ_MESH );
+  if(ends_with(inputDBName_, "bp")) {
```

Comment:
I would recommend that instead of depending on some extension type, we allow users to provide a `mesh_db_type: adios` and use that to read files, and set the default `mesh_db_type: exodus`